golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -101,12 +102,42 @@ func (k Keeper) GetReceiverImpl() vibctypes.ReceiverImpl {
 	return k
 }
 
+// Extract the base address from the packet sender (if senderIsLocal) or
+// receiver (if !senderIsLocal), since the local ibcModule doesn't understand
+// address parameters.
+func (k Keeper) packetWithOnlyBaseAddresses(packet channeltypes.Packet, senderIsLocal bool) channeltypes.Packet {
+	transferData := transfertypes.FungibleTokenPacketData{}
+	if err := k.cdc.UnmarshalJSON(packet.GetData(), &transferData); err != nil {
+		return packet
+	}
+	if senderIsLocal {
+		baseSender, err := types.ExtractBaseAddress(transferData.Sender)
+		if err == nil {
+			transferData.Sender = baseSender
+		}
+	} else {
+		baseReceiver, err := types.ExtractBaseAddress(transferData.Receiver)
+		if err == nil {
+			transferData.Receiver = baseReceiver
+		}
+	}
+
+	// Re-serialize the packet data with the base addresses.
+	data, _ := k.cdc.MarshalJSON(&transferData)
+
+	// Replace the packet's data with the new data.
+	packet.Data = data
+	return packet
+}
+
 // InterceptOnRecvPacket runs the ibcModule and eventually acknowledges a packet.
 // Many error acknowledgments are sent synchronously, but most cases instead return nil
 // to tell the IBC system that acknowledgment is async (i.e., that WriteAcknowledgement
 // will be called later, after the VM has dealt with the packet).
 func (k Keeper) InterceptOnRecvPacket(ctx sdk.Context, ibcModule porttypes.IBCModule, packet channeltypes.Packet, relayer sdk.AccAddress) ibcexported.Acknowledgement {
-	ack := ibcModule.OnRecvPacket(ctx, packet, relayer)
+	// Pass every (stripped-receiver) inbound to the wrapped IBC module.
+	strippedPacket := k.packetWithOnlyBaseAddresses(packet, false)
+	ack := ibcModule.OnRecvPacket(ctx, strippedPacket, relayer)
```

Comment:
```suggestion
	// Pass every (stripped-receiver) inbound packet to the wrapped IBC module.
	k.stripAddressDecoration(packet, RoleReceiver)
	ack := ibcModule.OnRecvPacket(ctx, packet, relayer)
```

Reply:
The original packet (`packet`) must be preserved for use later in this function.  `strippedPacket` must be a different struct (with different `strippedPacket.Data`) than `packet` or it breaks the `ibc_middleware_test.go` requirements.